### PR TITLE
chore: Update build scripts in `package.json` and code to get URL in `Party.tsx` as part of the deployment

### LINKED ISSUE #10: chore: Add Partykit as part of the deployment.
1. Run `npx partykit@latest token generate` to generate a token.
2. Run `ntl env:set PARTYKIT_LOGIN nickytonline` to add it to the environment variables for Netlify deploys.
3. Run `ntl env:set PARTYKIT_TOKEN THE_GENERATED_SUPER_SECRET_TOKEN` to add it to the environment variables for Netlify deploys.
4. Run `ntl env:set PUBLIC_PARTYKIT_HOST https://astro-partykit-starter-party.nickytonline.partykit.dev` so that the production URL for Partykit is available for the deployment.
5. Need to run `partykit deploy` as part of the deployment.

https://github.com/nickytonline/astro-partykit-starter/blob/f8f7cbb7d9f52c0d861e0027ef67661bffb1a6a7/package.json#L8

```diff
-  "build": "astro check && astro build",
+  "build": "astro check && astro build && partykit deploy",
```

6. Update the code for getting the Partykit URL to the following in https://github.com/nickytonline/astro-partykit-starter/blob/f8f7cbb7d9f52c0d861e0027ef67661bffb1a6a7/src/components/Party.tsx#L5

```diff
- const host = import.meta.env.PUBLIC_PARTYKIT_HOST ?? "localhost:1999";
+ const isProd = import.meta.env.PROD;
+ const host = isProd ? import.meta.env.PUBLIC_PARTYKIT_HOST : "localhost:1999";
```

### PR DESCRIPTION
**Pull request by adiati98**
## Description

This PR holds below changes:

- Update the build scripts to include `partykit deploy` in `package.json`.
- Update the code for getting the Partykit URL in `Party.tsx`.

## What type of PR is this? (check all applicable)

- [ ] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [x] 📦 Chore
- [ ] ⏩ Revert

## Related Tickets & Documents

Closes #10 

## Added to documentation?

- [ ] 📜 README.md
- [ ] 📜 CONTRIBUTING.md
- [x] 🙅 no documentation needed

### CODE REVIEW

**Comment by adiati98** [comment]
@nickytonline is there anything I should run locally to test this out?

**Comment by nickytonline** [comment]
When you run `npm run build` you should see it build successfully, including partykit.

If you get a build error, you may need to initialize partykit for yourself. See their docs, https://docs.partykit.io/.

To see if this is working on deploys, all the checks should have passed and if you go to the deploy preview, you should see your messages appear if you send a message.

**Comment by adiati98** [comment]
@nickytonline I have a couple of questions and issues:

- I ran `npm install` but I didn't commit changes in `.husky/_/husky.sh` and `package-lock.json`. Should I commit them?
- The deploy/netlify status is: `Expected — Waiting for status to be reported`. So I'm not sure if the checks are actually passed or not.
- I ran `npm run build` and got no error. So I proceeded with `npm start`. I could submitted username and room id, but when I enter message and hit the submit button, it didn't print out the message. What kind of behaviour do we expect here?
  
  partykit live preview

**Review by Satoshi-Sh — APPROVED** [review]
The updates look good :+1: 

About the unsent messages, the server is not running on the npm run dev.
Screenshot from 2023-12-31 09-29-10

```"dev": "partykit dev & astro dev",``` in the package.json is missing "&". 
Once I replaced it with ```"dev": "partykit dev && astro dev",```. It's all good

I think you can update it in this PR.

**Comment by adiati98** [comment]
@Satoshi-Sh good catch! Thanks! 👍

Just wondering, in which localhost did you run it locally?

I've added the `&`. And when I run `npm run dev`, I clicked "b" to open a browser as stated in the terminal:

npm run dev 404 not found

But I got "Not found" as below:

localhost 1999

This behaviour is the same with if there is a single `&` in the "dev" scripts.
That was the reason I ran `npm start` because we also have "start" in the scripts and open it on `http://localhost:4321/`.

**Comment by nickytonline** [comment]
Good catch @Satoshi-Sh. I was running it separately when I first created the project.

**Comment by Satoshi-Sh** [comment]
Screenshot from 2023-12-31 11-16-43

`npm run dev` will run client and server, so you just need to open localhost:4321 on your browser.  @adiati98

**Comment by adiati98** [comment]
@Satoshi-Sh @nickytonline I added the missing `&` and pushed the change.

---

I don't get the result as Satoshi had in my terminal when running `npm run dev`.

### Local Environment

Here how set up the local environment:

- Run `npm install`.
- Run `npm run dev`.

Is there any missing steps here?

### Behavior

1. After 10 minutes running `npm run dev`, nothing happened. Here is the screenshot: 

    npm run dev warning 1

    The `[pk:inf] GET / 404 Not Found (14ms)` was an ettempt to run `localhost: 1999` by typing "b" to open a browser after 10 minutes.

2. Then I clicked `[x] to exit` and got this:

    ready after x

3. So I opened the `http://localhost:4321/` and it still gave no print out. On console: `partysocket_react.js?v=ec93f343:399 WebSocket connection to 'ws://localhost:1999/party/01?_pk=8a2155a0-74b7-4b38-9f3f-4dfa61ef56ca' failed:` as screenshot below:

    console log partykit

**Comment by Satoshi-Sh** [comment]
It looks like && is different in Windows. It runs commands one by one. Once you stop the first command "party-kit dev", the second "astro dev" will run.  [referece](https://stackoverflow.com/questions/8055371/how-do-i-run-two-commands-in-one-line-in-windows-cmd#:~:text=%26%26%20%5B...%5D%0Acommand1%20%26%26%20command2%0AUse%20to%20run%20the%20command%20following%20%26%26%20only%20if%20the%20command%20preceding%20the%20symbol%20is%20successful.%20Cmd.exe%20runs%20the%20first%20command%2C%20and%20then%20runs%20the%20second%20command%20only%20if%20the%20first%20command%20completed%20successfully.)

I think "partykit dev & astro dev" will be working fine in Windows. 
I would put commands like this. 

```
"dev-win": "partykit dev & astro dev",
"dev-mac": "partykit dev && astro dev", 
```
Sorry for letting you @ayu back and forth.  

Let me know if you have another solution on this @nickytonline

**Comment by adiati98** [comment]
No worries at all, @Satoshi-Sh! 😄
Thanks for bearing with my long comments!!! 

I'll try the solution later on while waiting for @nickytonline response. 😊

**Comment by adiati98** [comment]
@Satoshi-Sh, an update. It behaves the same with double `&`.

**Comment by Satoshi-Sh** [comment]
I see. Then Windows users need to run the two commands in different terminals.
I would add `"start-server": "partykit dev",` to the scripts.

You can command `npm start` on one terminal and `npm run start-server` in another terminal.  

You can keep "dev": "partykit dev && astro dev", as is.

Is it better to add an instruction on how to run this application locally to README.md for Windows users? I believe Linux and Mac users can just run `npm run dev`.

**Comment by adiati98** [comment]
> I see. Then Windows users need to run the two commands in different terminals. I would add `"start-server": "partykit dev",` to the scripts.
> 
> You can command `npm start` on one terminal and `npm run start-server` in another terminal.
> 
> You can keep "dev": "partykit dev && astro dev", as is.
> 
> Is it better to add an instruction on how to run this application locally to README.md for Windows users? I believe Linux and Mac users can just run `npm run dev`.

This solution works!!!! 🎉

If @nickytonline agree with this, I can apply this and update the docs later. 👍

**Comment by adiati98** [comment]
@nickytonline @Satoshi-Sh 

If what we have in this PR now works for Mac and Linux, we can merge this as is. 👍

I will do some research and raise another issue to add solution for Windows.

**Comment by nickytonline** [comment]
Let's merge as is, and if you can create an issue for Windows @adiati98 we can get it sorted there. There's `nom-run-all` and `concurrently` packages that can do this. I think I'd probably go with `concurrently` based on this article. https://www.swyx.io/parallel-npm-scripts

File: package.json [hunk 1]
```diff
@@ -3,9 +3,9 @@
   "type": "module",
   "version": "0.0.1",
   "scripts": {
-    "dev": "partykit dev & astro dev",
+    "dev": "partykit dev && astro dev",
```
Comment (nickytonline) [inline package.json:6]: We'll sort this out in the Windows issue that will get created. Below works on non-Windows environments.

```suggestion
    "dev": "partykit dev & astro dev",
```

**Review by nickytonline — CHANGES_REQUESTED** [review]
This is looking good @adiati98. One small change request.

**Comment by adiati98** [comment]
> This is looking good @adiati98. One small change request.

I've fixed it, @nickytonline 👍

And I'll open an issue sometime today. 😊

**Comment by nickytonline** [comment]
I'm on vacation at the moment, but should get to testing this out before end of week.

**Review by nickytonline — APPROVED** [review]
Thanks @adiati98! 🚢

**Pull request merged**